[DEV] Add typecheck to lint workflow (#1210)
Despair-Games/poketernity: docs/linting.md

- **Consistency**: Ensures consistent coding style across the project.
- **Code Quality**: Helps catch potential errors and improve overall code quality.
- **Readability**: Makes the codebase easier to read and maintain.

# Biome

> We started transitioning over to [Biome](https://biomejs.dev/) for linting as it's performance far exceeds the performance of ESLint (~15x faster).
> 
> The idea is to have Biome replace ESLint completely in the future, but not all rules are supported yet so for now we're using both/a hybrid approach.

Reviewer: ```suggestion
We started transitioning over to [Biome](https://biomejs.dev/) for linting as its performance far exceeds ESLint's (~15x faster).

The end goal is to have Biome replace ESLint completely in the future, but not all rules are supported by Biome yet so for now we're using a hybrid approach.
```
